File: src/main/java/com/graphbrain/db/SourceNode.java
package com.graphbrain.db;

import java.util.HashMap;
import java.util.Map;

public class SourceNode extends Vertex {

	public SourceNode(String id) {
		super(id);
	}
  
	public SourceNode(String id, Map<String, String> map) {
		super(id);
	}
	
	@Override
	public VertexType type() {return VertexType.Source;}
	
	@Override
	public Map<String, String> toMap() {
		Map<String, String> map = new HashMap<String, String>();
		return map;
	}
}

File: core/src/main/scala/com/graphbrain/db/SourceNode.scala
package com.graphbrain.db

case class SourceNode(override val id: String,
                    override val degree: Int = 0,
                    override val ts: Long = -1)
  extends Vertex (id, degree, ts) {

  def this(id: String, map: Map[String, String]) =
    this(id, map("degree").toInt, map("ts").toLong)

  override def extraMap: Map[String, String] = Map()
}